### PR TITLE
Homework 8 completed

### PR DESCRIPTION
**Pull request by akulebyakin**
JDI

### CODE REVIEW

File: src/test/java/hw8/testdata/JdiSiteDataProvider.java [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+                    "Metal: " + item.getMetals(),
+                    "Vegetables: " + item.getVegetables()
+            };
+            result.add(new Object[]{item, expected});
+        }
+
+        return result.toArray(new Object[0][0]);
+    }
+
+    private ArrayList<MetalsAndColors> getMetalsAndColorsDataFromJson(String filepath)
```
Comment (troe88) [inline src/test/java/hw8/testdata/JdiSiteDataProvider.java:66]: Take a look at the Gson library and TypeToken specifically

File: src/test/java/hw8/testdata/FormData.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package hw8.testdata;
+
+import lombok.Data;
+
+import java.util.ArrayList;
+
+@Data
+public class FormData {
```
Comment (troe88) [inline src/test/java/hw8/testdata/FormData.java:8]: What kind of FormData is this ? What is the reason of two similar classes (MetalsAndColors ) ?

File: src/test/java/hw8/tests/JdiSiteTest.java [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import hw8.site.enums.HeaderMenuOptions;
+import hw8.site.pages.MetalsAndColorsPage;
+import hw8.testdata.JdiSiteDataProvider;
+import org.testng.annotations.Test;
+
+import static hw8.site.JdiSite.*;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+
+public class JdiSiteTest implements TestsInit {
```
Comment (troe88) [inline src/test/java/hw8/tests/JdiSiteTest.java:14]: Typical example of overdesign. Take a look at the task, you should create only ONE test with ONE dataprovider. It is not make sense at all to use DP everywhere.

File: src/test/java/hw8/tests/JdiSiteTest.java [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+    @Test(priority = 2, dataProviderClass = JdiSiteDataProvider.class,
+            dataProvider = "getTestUser")
+    public void checkUserLoggedIn(User user) {
+        homePage.userIcon.click();
+        loginForm.loginAs(user);
+
+        String userName = homePage.getCurrentUsername();
+
+        homePage.userName.assertThat().displayed();
```
Comment (troe88) [inline src/test/java/hw8/tests/JdiSiteTest.java:30]: You can use JDI assertions like ement.is().text("expected text")

File: src/test/java/hw8/tests/JdiSiteTest.java [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+    public void checkOpenMetalAndTransfer() {
+        header.headerMenu.select(HeaderMenuOptions.MetalsAndColors);
+        metalsAndColors.isOpened();
+    }
+
+    @Test(priority = 4, dataProviderClass = JdiSiteDataProvider.class,
+            dataProvider = "getMetalsAndColorsFormData")
+    public void checkMetalsAndColorsFill(MetalsAndColors formData, String[] expected) {
+        metalsAndColors.isOpened();
+        MetalsAndColorsPage.metalsAndColorsForm.fill(formData);
```
Comment (troe88) [inline src/test/java/hw8/tests/JdiSiteTest.java:44]: It will be better to create fomr::submit(entity) method instead of 2 methods (fill and submit)

**Review by troe88 — CHANGES_REQUESTED** [review]

File: src/test/java/hw8/testdata/FormData.java (hunk 2 above)
Comment (akulebyakin) [reply src/test/java/hw8/testdata/FormData.java:8]: Oops... I don't know how it got to the final cut.
Fixed.

File: src/test/java/hw8/tests/JdiSiteTest.java (hunk 5 above)
Comment (akulebyakin) [reply src/test/java/hw8/tests/JdiSiteTest.java:44]: Agree. Done!

File: src/test/java/hw8/tests/JdiSiteTest.java (hunk 4 above)
Comment (akulebyakin) [reply src/test/java/hw8/tests/JdiSiteTest.java:30]: Done

File: src/test/java/hw8/tests/JdiSiteTest.java (hunk 3 above)
Comment (akulebyakin) [reply src/test/java/hw8/tests/JdiSiteTest.java:14]: Changed it to just one test

File: src/test/java/hw8/testdata/JdiSiteDataProvider.java (hunk 1 above)
Comment (akulebyakin) [reply src/test/java/hw8/testdata/JdiSiteDataProvider.java:66]: Done

**Review by troe88 — APPROVED** [review]

**Pull request merged**